### PR TITLE
MistypedExponentiation: Add a heuristic to reduce FPs

### PR DESCRIPTION
**Pull request by sauyon**
Fixes two FPs on LGTM.com.

### CODE REVIEW

File: ql/src/InconsistentCode/MistypedExponentiation.ql [hunk 1]
```diff
@@ -32,6 +32,9 @@ where
     exists(Ident id | id = xe.getRightOperand() |
       id.getName().regexpMatch("(?i)_*((exp(onent)?)|pow(er)?)")
     )
+  ) and
+  not exists(Assignment assign | assign.getRhs() = xe.getParent*() |
+    assign.getLhs().getAChild*().(Ident).getName().regexpMatch(".*(^m|M)ask($|\\p{Lu}).*")
```
Comment (max-schaefer) [inline ql/src/InconsistentCode/MistypedExponentiation.ql]: Could you add a comment explaining the heuristic and a test, please?

**Comment by sauyon** [comment]
Added a comment and a test.

File: ql/src/InconsistentCode/MistypedExponentiation.ql (hunk 1 above)
Comment (max-schaefer) [reply ql/src/InconsistentCode/MistypedExponentiation.ql]: I don't think we'd lose much by simplifying the regex to `(?i).*mask.*`. Also, can you restrict it to the case where the LHS is either an identifier or a selector? All those transitive closures make me a bit uneasy.

Comment (sauyon) [reply ql/src/InconsistentCode/MistypedExponentiation.ql]: Ok, I made the changes.

**Comment by max-schaefer** [comment]
Thanks. Since this is a user-visible change, can you add a change note as
well, please?

**Pull request merged**